qgis_deployment_toolbelt/jobs/job_qgis_installation_finder.py

```diff
@@ -234,53 +260,83 @@ def _get_windows_installed_qgis_path() -> dict[str, str]:
-            version_str = JobQgisInstallationFinder._get_qgis_bin_version(
-                qgis_bin=qgis_bin
-            )
-            if version_str:
-                found_version[version_str] = qgis_bin
-            else:
-                logger.warning(f"Can't define QGIS version for '{qgis_bin}' binary.")
+        search_paths.append(environ.get("QDT_OSGEO4W_INSTALL_DIR", "C:\\OSGeo4W"))
+
+        # search_paths
+        option_search_path = self._get_search_paths_with_environment_variable()
+        if len(option_search_path) != 0:
+            search_paths = option_search_path + search_paths
+
+        return JobQgisInstallationFinder._get_qgis_found_version_dict_from_search_paths(
+            search_paths=search_paths
+        )
+
+    @staticmethod
+    def _get_qgis_found_version_dict_from_search_paths(
+        search_paths: list[str],
+    ) -> dict[str, str]:
+        """Define qgis found version dict from a list of search path
+        If identical version are found in multiple path, the first version found in search_path is used.
+
+        Args:
+            search_paths (list[str]): list of search paths
+
+        Returns:
+            dict[str, str]: dict of qgis binary path for qgis version
+        """
+        # We search reversed to have the version defined in priority with the first value
+        found_version = {}
+        for search_path in reversed(search_paths):
+            if qgis_bin := JobQgisInstallationFinder._get_qgis_bin_in_install_dir(
+                search_path
+            ):
+                JobQgisInstallationFinder._search_qgis_version_and_add_to_dict(
+                    qgis_bin=qgis_bin, found_version=found_version
+                )
 
         return found_version
 
     @staticmethod
-    def _get_linux_installed_qgis_path() -> dict[str, str]:
+    def _search_qgis_version_and_add_to_dict(
+        qgis_bin: str, found_version: dict[str, str]
+    ) -> None:
+        """Search qgis version from qgis binary and add to found_version dict if found
+
+        Args:
+            qgis_bin (str): qgis binary path
+            found_version (dict[str, str]): updated dict of qgis binary path and version
+        """
+        version_str = JobQgisInstallationFinder._get_qgis_bin_version(qgis_bin=qgis_bin)
+        if version_str:
+            found_version[version_str] = qgis_bin
+        else:
+            logger.warning(f"Can't define QGIS version for '{qgis_bin}' binary.")
+
+    def _get_linux_installed_qgis_path(self) -> dict[str, str]:
         """Get install qgis path for linux operating system with which
 
         Returns:
             dict[str, str]: dict of QGIS version and QGIS bin path
         """
+
+        # search_paths
+        search_paths = self._get_search_paths_with_environment_variable()
+
+        found_version = (
+            JobQgisInstallationFinder._get_qgis_found_version_dict_from_search_paths(
+                search_paths=search_paths
+            )
+        )
```

Comment:
This won't work on Linux because we are searching for QGIS with a executable pattern : `qgis(-ltr)?-bin\.exe`

@Guts  do we need to support `search_paths` option on Linux ? I don't think we will have the case where multiple QGIS are installed on Linux.

Reply:
Not so sure. There is a certain number of people using conda or nix to have multiple QGIS versions installed in //.

Reply:
Same behavior as Windows now but we search for `qgis` and not `qgis-bin.exe` or `qgis-ltr-bin.exe`